st2common/st2common/services/executions.py

```diff
@@ -136,10 +137,29 @@ def update_execution(liveaction_db, publish=True):
     return execution
 
 
+def abandon_execution_if_incomplete(liveaction_id, publish=True):
+    """
+    Marks execution as abandoned if it is still incomplete. Abandoning an
+    execution implies that its end state is unknown and cannot anylonger
+    be determined. This method should only be called if the owning process
+    is certain it can no longer determine status of an execution.
+    """
+    liveaction_db = action_utils.get_liveaction_by_id(liveaction_id)
+    # No need to abandon and already complete action
+    if liveaction_db.status in action_constants.LIVEACTION_COMPLETED_STATES:
+        return
+    liveaction_db = action_utils.update_liveaction_status(
+                status=action_constants.LIVEACTION_STATUS_ABANDONED,
```

Comment:
Yeah that just work since it is eventually an execution being handled by some action-runner. If the action-runner handling the child execution is still around and processing then that execution will succeed.